### PR TITLE
[Merged by Bors] - feat: another composition of `ContinuousOn` and `Continuous`

### PR DESCRIPTION
**Pull request by scholzhannah**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/ContinuousOn.lean [hunk 1]
```diff
@@ -861,6 +861,10 @@ theorem ContinuousOn.comp_continuous {g : β → γ} {f : α → β} {s : Set β
   rw [continuous_iff_continuousOn_univ] at *
   exact hg.comp hf fun x _ => hs x
 
+theorem ContinuousOn.image_comp_continuous {g : β → γ} {f : α → β} {s : Set α}
+    (conton : ContinuousOn g (f '' s)) (cont : Continuous f) : ContinuousOn (g ∘ f) s :=
```
Comment (grunweg) [inline Mathlib/Topology/ContinuousOn.lean:865]: I think naming the hypotheses `hg` and `hf`, respectively, would be more in line with the other lemmas in the file.

**Review by grunweg — COMMENTED** [review]
Thanks! I wonder if this lemma is worth having (as it's a combination of just a few small lemmas). Do you have an application for it? Do you, for example, need it in one of your projects depending on mathlib?

Apart of one naming nit, the code itself looks good to me.

**Comment by scholzhannah** [comment]
> Thanks! I wonder if this lemma is worth having (as it's a combination of just a few small lemmas). Do you have an application for it? Do you, for example, need it in one of your projects depending on mathlib?

I use it when precomposing the characteristic maps of a CW-complex, for example to construct the product of two CW-complexes. But I also have only used that lemma three times. So if it is not useful anywhere else, I don't mind not having it.

**Comment by grunweg** [comment]
For me, three times is useful enough: let's get a maintainer's opinion on this:
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**